### PR TITLE
add support for renamed files to updateconfig

### PR DESCRIPTION
**Pull request by BenTheElder**
fixes: https://github.com/kubernetes/test-infra/issues/8891

/area prow

### CODE REVIEW

File: prow/github/types.go [hunk 1]
```diff
@@ -214,14 +214,15 @@ const (
...
-	BlobURL   string `json:"blob_url"`
+	SHA              string `json:"sha"`
+	Filename         string `json:"filename"`
+	Status           string `json:"status"`
+	Additions        int    `json:"additions"`
+	Deletions        int    `json:"deletions"`
+	Changes          int    `json:"changes"`
+	Patch            string `json:"patch"`
+	BlobURL          string `json:"blob_url"`
+	PreviousFilename string `json:"previous_filename"`
```
Comment (cjwagner) [inline prow/github/types.go]: Nice, this field is not only undocumented, but not even referenced in the example response payload in GitHub's docs: https://developer.github.com/v3/pulls/#list-pull-requests-files

(Are we sure it is real?)

Comment (krzyzacy) [reply prow/github/types.go]: yeah we did an actual api call... their example does not list all valid status :-(

Comment (BenTheElder) [reply prow/github/types.go]: lol, I got it in debugging https://github.com/kubernetes/test-infra/issues/8891, I think we can probably depend on it? most of github's API fields are not properly documented, FWIW.

**Pull request merged**